### DISCUSSION TITLE
No Video pass through to client device when using virtual monitor

### DISCUSSION THREAD

**Discussion by QuadT** [ClassicOldSong/Apollo#110] [Q&A]
Having a weird situation with the 0.2.6-Aplha2 build. First time trying out Apollo so its probably something I'm doing wrong. It appears to install fine and the sudamaker VDA shows up in my device manager but everytime I try to stream to a moonlight device I only get sound and no video. if I turn on my monitor I can see what should be sent to the client device. Any idea what's going on here?

**Comment by ClassicOldSong** [comment]
1. Make sure you start streaming using the Virtual Desktop entry.
2. Make sure you're using Extended display mode or Only display on 2 in Windows display settings.
3. There's no need to turn off your monitor manually, if you use Only display on 2, Windows will turn off the monitor automatically.
4. You can enable Headless Mode in Audio/Video tab, that makes everything start in Virtual Display even without a monitor connected.

If you're still not getting any image, please share your system information and I'll try to see what's wrong.

**Reply by QuadT** [reply]
I had it working for a little bit yesterday by using the Virtual Desktop entry and dragging playnite over to it what became the 3rd screen and making the app fullscreen. However a few hours later I went to use it again and it was back to not working. The only difference would be that I woke the host machine up from sleep with the moonlight app and the primary monitor was no longer on.

System Info:
MSI Carbon x870e
AMD 9800X3D
64 GB RAM 6000MT/s
Nvidia 4090
main screen is an LG 48 inch C4 tv (HDMI)
secondary screen is some old Asus 2560x1440 in portrait mode (displayport)
Windows 11 24H2

I appreciate you taking time to look into this :)

**Reply by ClassicOldSong** [reply]
If you're seeing a black screen on login UI and the virtual display isn't primary, that's expected. Windows only show login on primary display.

**Comment by QuadT** [comment]
So, I did set the Virtual Display as primary and watched it take over a
couple times before walking away for a few hours. Then when I tried again
and woke the host device up it was back to black screen on the client side.

**Reply by ClassicOldSong** [reply]
You mentioned that you have two physical monitors, that might cause problems when one of the goes to sleep/disconnect, in that case the display configuration changed, so Windows might changed the primary display back to one of the other physical display that is still awake.

**Comment by QuadT** [comment]
That is something I was thinking about as well, yeah... I'll need to do a
bit of testing when I get home tonight.

**Comment by QuadT** [comment]
Ok, I think I solved the problem. After a bunch of failed attempts I was
looking through the logs and noticed that it kept referencing an AMD
Display driver. So I disabled the IGPU in the BIOS and uninstalled the AMD
Adrenaline software. Boom, works like a charm now. It was trying to use the
wrong GPU for the output.

With the multi-monitor aspect I just set screen 3 (virtual display) as the
primary while I was streaming to a client and moved the playnite app over
to that one. Now when I launch things in moonlight it all comes up fine,
even after being asleep. It even picks up the virtual display for me to put
my windows PIN in without issue.

Thanks again!

QuadT

**Reply by ClassicOldSong** [reply]
Ah, in this case you can just set the adapter name to your dgpu, but you won't be able to stream displays connected to your igpu then. But it seems you're not using igpu anyway so it should be fine.

**Comment by QuadT** [comment]
Yeah, since I'm not using it I'm fine with disabling it. Thanks again!
